index.js

```diff
@@ -60,13 +60,48 @@ animalToRemove.remove()
 
 /***** Deliverable 1 *****/
 function toggleColor(element) {
-    if (element.style.color === "green") {
+    if (element.style.color === "red") {
         element.style.color = "black"
     } else {
-        element.style.color = "green"
+        element.style.color = "red"
     }
 }
+header.addEventListener('click', function(event) {
+    toggleColor(event.target)
+})
 
 /***** Deliverable 2 *****/
 
+const increaseLike = document.querySelector('.like-button')
+increaseLike.addEventListener('click', function() {
+    
+    likes.textContent = `${traveler.likes += 1} Likes`
+})
+
 /***** Deliverable 3 *****/
+
+const newAnimalForm = document.querySelector('#new-animal-sighting-form')
+
+newAnimalForm.addEventListener('submit', function(event) {
+    event.preventDefault()
+    
+    const inputFieldOne = event.target.species.value
+    const inputFieldTwo = event.target.link.value
+    const inputFieldThree = event.target.photo.value
+    const inputFieldFour = event.target.description.value
+    const animalId = traveler.animalSightings[traveler.animalSightings.length - 1].id + 1
+
+
+    const newSighting = {
+        species: inputFieldOne,
+        photo: inputFieldTwo,
+        link: inputFieldThree,
+        description: inputFieldFour,
+        id: animalId,
+        travelerId: 8
+    }
+    traveler.animalSightings.push(newSighting)
+    renderAnimalSightingPost(newSighting)   
+    //photo not showing on page?? Chrome error of some sort
```

Comment:
Great job here! Again, I would create a separate function to be passed in as a callback function. I would make the variable names more descriptive than "inputFieldOne", and change it to something descriptive of what it is, which is a string of information about an animal `animalSpecies`.

Great job assembling an object of the data! Also impressed that you were thinking about adding the ID!!! 🤯 When we work with posting to the backend, we will be creating a new instance with rails which you know will automatically create an id for us! But this is awesome :)

You are thinking ahead with line 103, but we will have a more dynamic way of persisting the data to the backend. stay tuned :)

Great job using the pre-existing function on line 104!

Don't forget to reset the form to clear out the input fields!

My steps for solving this are as follows:

1. Find the sightingForm on the page and save it to a variable
2. add an event listener to the form element, listen for a `submit` event.
3. The submit event needs to be applied to an entire form element, not the submit button.
4. Create the `handleFormSubmission` callback function, which should be triggered when the form submits.

`handleFormSubmission` steps:

1. since we are submitting a form, we need to prevent the default action of the form, which is to make a post request and redirect to a specific URL. We do not want the page to refresh, we want to prevent that action and stay on the current page.
2. save all the data the user typed in from the form into descriptive variables.
3. assemble an object with key value pairs descriptive of the data they contain.
4. Pass in the animal data object to the `renderAnimalSightingPost` function

```
const sightingForm = document.querySelector(`form#new-animal-sighting-form`);

function handleFormSubmission(e){
    e.preventDefault();
    const species = e.target["species"].value;
    const videoLink = e.target["animal-link"].value;
    const photo = e.target["animal-photo"].value;
    const description = e.target["sighting-description"].value;
    const newData = {
        species: species,
        videoLink: videoLink,
        photo: photo,
        description: description
    }
    renderAnimalSightingPost(newData);
}

sightingForm.addEventListener("submit", handleFormSubmission)
```

With ES6 notation

```
const sightingForm = document.querySelector(`form#new-animal-sighting-form`);

const handleFormSubmission = e => {
    e.preventDefault();
    const species = e.target["species"].value;
    const videoLink = e.target["animal-link"].value;
    const photo = e.target["animal-photo"].value;
    const description = e.target["sighting-description"].value;
    const newData = {
        species,
        videoLink,
        photo,
        description
    }
    e.target.reset()
    renderAnimalSightingPost(newData);
}

sightingForm.addEventListener("submit", handleFormSubmission)
```